src/main/java/in/ac/bits/protocolanalyzer/persistence/repository/SaveRepository.java

```diff
@@ -14,44 +16,49 @@
 import com.google.common.eventbus.EventBus;
 import com.google.common.eventbus.Subscribe;
 
-import javax.naming.Context;
-import javax.naming.InitialContext;
-import javax.naming.NamingException;
-
 import in.ac.bits.protocolanalyzer.analyzer.event.BucketLimitEvent;
 import in.ac.bits.protocolanalyzer.analyzer.event.EndAnalysisEvent;
 import in.ac.bits.protocolanalyzer.analyzer.event.SaveRepoEndEvent;
+import lombok.extern.log4j.Log4j;
 
 @Component
+@ComponentScan("config.in.ac.bits.protocolanalyzer.persistence.repository")
 @Scope("prototype")
 @Log4j
 public class SaveRepository implements Runnable {
 
 	@Autowired
 	private ElasticsearchTemplate template;
-
+	
+	@Autowired
 	private ConcurrentLinkedQueue<ArrayList<IndexQuery>> buckets;
-
+	
+	@Autowired
+	private Context ctx;
+	
+	@Autowired
+	private Runtime runtime;
+	
+	public long memory;
+	
 	private boolean isRunning = false;
 
-	private boolean analysisRunning = true;
```

Comment:
It is better to keep the fields private. If access to the field is required for resting, we can use the available getter method.